docs/source/graphs.rst

```diff
@@ -97,3 +97,34 @@ special and others can write different schedulers better suited to other
 applications or architectures easily.  Systems that emit dask graphs (like
 Dask Array, Dask Bag, and so on) may leverage the appropriate scheduler for
 the application and hardware.
+
+
+Task Expectations
+-----------------
+
+When a task is submitted to Dask for execution, there are a number of assumptions
+that are made about that task. In general, tasks with side-effects that alter the
+state of a future in-place are not recommended. Values stored by Dask are mutable,
+and can be updated in-place. For example, consider a workflow involving a
+``np.array``:
```

Comment:
Technically the term here is `np.ndarray`, but I suspect that the following will be more universally understood.

```suggestion
Numpy array:
```